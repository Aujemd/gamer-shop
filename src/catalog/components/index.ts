export { default as CatalogHeader } from "./catalog-header";
