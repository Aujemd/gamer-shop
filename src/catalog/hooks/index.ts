export { useCatalogProducts } from "./use-catalog-products";
