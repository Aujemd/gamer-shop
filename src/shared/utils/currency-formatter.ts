/**
 * Formats a number as a currency string.
 * @param amount - The amount to format.
 * @param locale - The locale to use for formatting (default is "en-US").
 * @param currency - The currency code to use for formatting (default is "USD").
 * @returns The formatted currency string.
 */
export function formatCurrency(amount: number, locale: string = "en-US", currency: string = "USD"): string {
    return new Intl.NumberFormat(locale, {
        style: "currency",
        currency,
    }).format(amount);
}
