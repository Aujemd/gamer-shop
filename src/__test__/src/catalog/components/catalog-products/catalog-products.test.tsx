import { render, screen } from "@testing-library/react";

import { CatalogProducts } from "@/catalog";
import { allGames } from "@/utils/endpoint";

const mockProducts = allGames;

describe("catalogProducts", () => {
    it("should render correctly", () => {
        const { asFragment } = render(<CatalogProducts products={mockProducts} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("applies custom classname", () => {
        render(<CatalogProducts products={mockProducts} className="custom-class" data-testid="catalog-products-test-id" />);
        const section = screen.getByTestId("catalog-products-test-id");
        expect(section).toHaveClass("custom-class");
    });
    it("renders a list of product cards", () => {
        render(<CatalogProducts products={mockProducts} />);

        mockProducts.forEach((product) => {
            expect(screen.getByTestId(`catalog-products-ml-product-card-${product.id}-test-id`)).toBeInTheDocument();
        });
    });
    it("renders a 'See More' button", () => {
        render(<CatalogProducts products={mockProducts} />);
        const seeMoreButton = screen.getByTestId("catalog-products-see-more-button-test-id");
        expect(seeMoreButton).toBeInTheDocument();
        expect(seeMoreButton).toHaveTextContent("See More");
    });
});
