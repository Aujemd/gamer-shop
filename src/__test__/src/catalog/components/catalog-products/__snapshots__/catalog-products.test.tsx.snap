// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catalogProducts should render correctly 1`] = `
<DocumentFragment>
  <section
    products="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
  >
    <div
      class="max-w-desktop mx-auto"
    >
      <ul
        class="grid grid-cols-1 gap-6 py-8 px-6 md:grid-cols-2 lg:grid-cols-3 lg:gap-12 xl:pl-0 xl:pr-11 xl:py-12 items-stretch "
      />
      <svg
        class="w-8 fill-cta-fill-primary mx-auto my-6"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <circle
            cx="12"
            cy="3"
            r="1"
          >
            <animate
              attributeName="r"
              begin="0;spinner_tKsu.end-0.5s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_7Z73"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="16.50"
            cy="4.21"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_7Z73.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_Wd87"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="7.50"
            cy="4.21"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_9Qlc.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_tKsu"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="19.79"
            cy="7.50"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_Wd87.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_lMMO"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="4.21"
            cy="7.50"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_Khxv.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_9Qlc"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="21.00"
            cy="12.00"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_lMMO.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_5L9t"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="3.00"
            cy="12.00"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_ld6P.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_Khxv"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="19.79"
            cy="16.50"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_5L9t.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_BfTD"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="4.21"
            cy="16.50"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_XyBs.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_ld6P"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="16.50"
            cy="19.79"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_BfTD.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_7gAK"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="7.50"
            cy="19.79"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_HiSl.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_XyBs"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <circle
            cx="12"
            cy="21"
            r="1"
          >
            <animate
              attributeName="r"
              begin="spinner_7gAK.begin+0.1s"
              calcMode="spline"
              dur="0.6s"
              id="spinner_HiSl"
              keySplines=".27,.42,.37,.99;.53,0,.61,.73"
              values="1;2;1"
            />
          </circle>
          <animatetransform
            attributeName="transform"
            dur="6s"
            repeatCount="indefinite"
            type="rotate"
            values="360 12 12;0 12 12"
          />
        </g>
      </svg>
    </div>
  </section>
</DocumentFragment>
`;
